proposals/lip-0054.md

```diff
@@ -368,19 +392,21 @@ def verify(trs: Transaction) -> None:
     if len(trsParams.idxs) != len(trsParams.crossChainMessages):
         raise Exception("Inclusion proof indices and cross-chain messages do not have the same length.")
 
-    # Check that the idxs are sorted in ascending order.
-    if trsParams.idxs != sorted(trsParams.idxs):
-        raise Exception("Cross-chain message indexes are not sorted in ascending order.")
-
-    # Check that the idxs are unique.
-    if len(trsParams.idxs) != len(set(trsParams.idxs)):
-        raise Exception("Cross-chain message indexes are not unique.")
+    # Check that the idxs are strictly increasing
+    for i in range(len(trsParams.idxs)-1):
+        if not (trsParams.idxs[i] < trsParams.idxs[i+1]):
+            raise Exception("Cross-chain message indexes are not strictly increasing.")
     
-    # Check that the CCMs are still pending. We can check only the first one,
+    # Check that the CCMs are still pending. We check only the first one,
     # as the idxs are sorted in ascending order.
     if trsParams.idxs[0] < terminatedOutboxAccount(trsParams.chainID).partnerChainInboxSize:
         raise Exception("Cross-chain message is not pending.")
 
+    # Check that the CCM indices do not exceed the outbox size. We check only the last one,
+    # as the idxs are sorted in ascending order.
+    if terminatedOutboxAccount(trsParams.chainID).outboxSize <= trsParams.idxs[-1]:
```

Comment:
Why is it `<=` and not just `<` ?

Reply:
I think it should be `<=` but please double check

Reply:
Example: Say `outboxSize = 10`. So, there are 10 messages in the outbox with the indices 0 to 9. If you try to recover a message with `index = outboxSize = 10`, then you try to recover the 11th message of the outbox, which does not exist.

Reply:
ah right, start from 0, sorry my bad. Ok, can be resolved from my side.